tox.ini

```diff
@@ -114,10 +114,12 @@ commands =
     mpldev: pip install -U --pre -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple matplotlib
 
     pip freeze
-    !cov-!double: pytest --pyargs astropy {toxinidir}/docs {env:MPLFLAGS} {posargs}
-    cov-!double: pytest --pyargs astropy {toxinidir}/docs {env:MPLFLAGS} --cov astropy --cov-config={toxinidir}/pyproject.toml {posargs}
+
+    !double: pytest --pyargs astropy {toxinidir}/docs {env:MPLFLAGS} \
+    cov: --cov astropy --cov-config={toxinidir}/pyproject.toml --cov-report xml:{toxinidir}/coverage.xml \
```

Comment:
We don't do `double-cov` but I also don't think we forbid it... will this work if someone pass in double-cov?

Reply:
I guess what I mean is this looks fine but I am not sure about removing `cov: coverage xml -o {toxinidir}/coverage.xml`. I could be overthinking this...

Reply:
Does `double-cov` make sense though?  Currently `double` doesn't run the tests twice -- that is handled in the Actions workflow, e.g. `test` followed by `test-double` jobs.  Why would someone want to do `test-cov` followed by `test-double-cov`?

Reply:
Does that mean it is safer to change `cov:` to `cov-!double:`?

Reply:
The `cov: coverage xml -o {toxinidir}/coverage.xml` line simply converts the coverage data (generated by pytest) into an xml file.  In this PR, that is done with `--cov-report xml:{toxinidir}/coverage.xml`

Reply:
| Does that mean it is safer to change cov: to cov-!double:?

It would have no effect because that entire `pytest` command already only runs if `!double`.

Reply:
I don't understand how this `\` works. It looks to me like `tox` would still call `cov:` line when `double` is passed in now and the `\` is only put together after `tox` collects all the command sub-string. Or is that not true?

Reply:
```
    !double: pytest --pyargs astropy {toxinidir}/docs {env:MPLFLAGS} \
    cov: --cov astropy --cov-config={toxinidir}/pyproject.toml --cov-report xml:
{toxinidir}/coverage.xml \
    {posargs}
```

is all *one*, dynamically-generated, command.

It starts with `!double`, so all of what follows only runs for `!double`.  If `cov` is also specified (i.e. `!double-cov`), then the flags on the `cov:` line get added.

So !double-cov runs every line:
`pytest --pyargs astropy {toxinidir}/docs {env:MPLFLAGS}  cov: --cov astropy --cov-config={toxinidir}/pyproject.toml --cov-report xml:{toxinidir}/coverage.xml {posargs}`

and 

!double-!cov runs:
`pytest --pyargs astropy {toxinidir}/docs {env:MPLFLAGS} {posargs}`

That's exactly what it did before, but broken into two separate pytest commands.

Reply:
My tox-fu is weak, so thanks for the clarification!

Reply:
>  Currently double doesn't run the tests twice

It does, just below it run `test()` twice.

Reply:
Ah yes, I see that it does run twice there - that's good.